src/pages/Signup/Signup.js

```diff
@@ -1,5 +1,237 @@
-import React from 'react';
+import React, { useState } from 'react';
+import TERMS_DATA from './SignupData';
+import './Signup.scss';
 
 export default function Signup() {
-  return <div>hi</div>;
+  const [memberInput, setMemberInput] = useState({
+    email: '',
+    password: '',
+    passwordRe: '',
+    address: '',
+    isVegan: false,
+  });
+  const [emailValid, setEmailValid] = useState(false);
+  const [passwordValid, setPasswordValid] = useState(false);
+  const [passwordReValid, setPasswordReValid] = useState(false);
+  const [addressValid, setAddressValid] = useState(false);
+  const [checkedList, setCheckedList] = useState([]);
+
+  const { email, password, passwordRe, address, isVegan } = memberInput;
+
+  const onCheckedAll = checked => {
+    const newCheckedList = checked ? TERMS_DATA.map(term => term.id) : [];
+    setCheckedList(newCheckedList);
+  };
+
+  const onCheckedElement = (checked, termId) => {
+    checked
+      ? setCheckedList([...checkedList, termId])
+      : setCheckedList(checkedList.filter(el => el !== termId));
```

Comment:
```suggestion
    const newCheckedList = checked ? [...checkedList, termId] : checkedList.filter(el => el !== termId);
    setCheckedList(newCheckedList);
```
setCheckedList에 인자로 넣는 값만 조건에 따라 변경되면 되기에 위와 같이 표현하는게 더 좋아보이네요! setCheckedList를 두번이나 써줄 필요는 없으니까요!